### DISCUSSION TITLE
How to create two tables from the same model

### DISCUSSION THREAD

**Discussion by jairoareyes** [fnc12/sqlite_orm#1138] [Q&A]
Hi friend, I'd like to congratulate you on this amazing framework. 

I'd like to know if it is possible to have two tables using the same class, and example could be: 

```
make_table<TestModel>("test_model_a",
    make_column("value", &TestModel::value)),

make_table<TestModel>("test_model_b",
    make_column("value", &TestModel::value)),
```

I have already tried defining the database and it seems to work, but I don't know how to differentiate each table in a query. Thanks in advance.

**Comment by trueqbit** [comment]
Hi @jairoareyes,

since the class type is used for building the "storage" as we call it and for lookup when building the SQL query, you cannot use the same class in two different tables.

What you can do is create at least one derived class to disambiguate.

**Comment by fnc12** [comment]
Hi Jairo. How do you expect to retrieve data with such an approach? E.g. you need to `get_all` models. When you will write `storage.get_all<TestModel>()` from which table you expect to obtain data?
